src/java.base/share/classes/java/util/concurrent/ThreadLocalRandom.java

```diff
@@ -1066,22 +422,55 @@ private Object readResolve() {
     private static final long INHERITEDACCESSCONTROLCONTEXT
         = U.objectFieldOffset(Thread.class, "inheritedAccessControlContext");
 
-    /** Rarely-used holder for the second of a pair of Gaussians */
-    private static final ThreadLocal<Double> nextLocalGaussian =
-        new ThreadLocal<>();
-
     /** Generates per-thread initialization/probe field */
     private static final AtomicInteger probeGenerator = new AtomicInteger();
 
     /** The common ThreadLocalRandom */
     static final ThreadLocalRandom instance = new ThreadLocalRandom();
 
+    private static final class ThreadLocalRandomProxy extends Random {
+        @java.io.Serial
+        static final long serialVersionUID = 0L;
+
+       public int nextInt() {
+           return ThreadLocalRandom.current().nextInt();
+       }
+
+       public long nextLong() {
+           return ThreadLocalRandom.current().nextLong();
+       }
+    }
+
+    private static final AbstractSpliteratorGenerator proxy = new ThreadLocalRandomProxy();
+
+    /**
+     * Returns a {@code RandomGenerator} object that uses {@code ThreadLocalRandom}.
+     * This object is threadsafe (methods of this object may be called by threads
+     * other than the current one).
+     *
+     * @return a {@code RandomGenerator} object that uses {@code ThreadLocalRandom}
+     */
+    public static RandomGenerator proxy() {
```

Comment:
proxy is a generic name, is there a better name here ?

Reply:
will investigate

Reply:
agree